### PR TITLE
airflow: add script for easier creation of local development environment

### PR DESCRIPTION
**Pull request by mobuchowski**
This script adds possibility to run any Airflow image with OpenLineage integrations build from current branch.

To do that, run it from airflow integration directory as 

```bash
AIRFLOW_IMAGE=<airflow_image_with_tag> ./scripts/run-airflow.sh
```

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

### CODE REVIEW

**Comment by sekikn** [comment]
It's a great improvement for testing! LGTM except for a few minor points:

* Running without the integration/python directory causes the following error.

  ```
  rm: cannot remove './python': No such file or directory
  ```

* The owner of the integration/airflow/scripts/airflow directory has to be consistent with the container image. For example, if apache/airflow:1.10.15-python3.7 is specified as AIRFLOW_IMAGE, users have to create a user with uid 50000 and assign it to the owner of that directory in advance. Otherwise they come across the following error:

  ```
  airflow_init_1         | Unable to load the config, contains a configuration error.
  airflow_init_1         | Traceback (most recent call last):
  airflow_init_1         |   File "/usr/local/lib/python3.7/logging/config.py", line 563, in configure
  airflow_init_1         |     handler = self.configure_handler(handlers[name])
  airflow_init_1         |   File "/usr/local/lib/python3.7/logging/config.py", line 736, in configure_handler
  airflow_init_1         |     result = factory(**kwargs)
  airflow_init_1         |   File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/log/file_processor_handler.py", line 50, in __init__
  airflow_init_1         |     os.makedirs(self._get_log_directory())
  airflow_init_1         |   File "/usr/local/lib/python3.7/os.py", line 213, in makedirs
  airflow_init_1         |     makedirs(head, exist_ok=exist_ok)
  airflow_init_1         |   File "/usr/local/lib/python3.7/os.py", line 223, in makedirs
  airflow_init_1         |     mkdir(name, mode)
  airflow_init_1         | PermissionError: [Errno 13] Permission denied: '/opt/airflow/logs/scheduler'
  airflow_init_1         |
  airflow_init_1         | The above exception was the direct cause of the following exception:
  airflow_init_1         |
  airflow_init_1         | Traceback (most recent call last):
  airflow_init_1         |   File "/home/airflow/.local/bin/airflow", line 25, in <module>
  airflow_init_1         |     from airflow.configuration import conf
  airflow_init_1         |   File "/home/airflow/.local/lib/python3.7/site-packages/airflow/__init__.py", line 47, in <module>
  airflow_init_1         |     settings.initialize()
  airflow_init_1         |   File "/home/airflow/.local/lib/python3.7/site-packages/airflow/settings.py", line 436, in initialize
  airflow_init_1         |     LOGGING_CLASS_PATH = configure_logging()
  airflow_init_1         |   File "/home/airflow/.local/lib/python3.7/site-packages/airflow/logging_config.py", line 68, in configure_logging
  airflow_init_1         |     raise e
  airflow_init_1         |   File "/home/airflow/.local/lib/python3.7/site-packages/airflow/logging_config.py", line 63, in configure_logging
  airflow_init_1         |     dictConfig(logging_config)
  airflow_init_1         |   File "/usr/local/lib/python3.7/logging/config.py", line 800, in dictConfig
  airflow_init_1         |     dictConfigClass(config).configure()
  airflow_init_1         |   File "/usr/local/lib/python3.7/logging/config.py", line 571, in configure
  airflow_init_1         |     '%r' % name) from e
  airflow_init_1         | ValueError: Unable to configure handler 'processor'
  ```

  I think it should be documented somewhere.

File: .circleci/get-docker-compose.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 set -e
 
-curl -L https://github.com/docker/compose/releases/download/1.25.3/docker-compose-`uname -s`-`uname -m` > ~/docker-compose
+curl -L https://github.com/docker/compose/releases/download/1.29.2/docker-compose-`uname -s`-`uname -m` > ~/docker-compose
```
Comment (JDarDagran) [inline .circleci/get-docker-compose.sh:9]: I've upgraded this to have `profiles` available in docker-compose

File: integration/airflow/scripts/run-dev-airflow.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# this makes it easier to keep proper permissions on logs
+mkdir -p airflow/logs
+chmod a+rwx -R airflow/logs
+
+# maybe overkill
+OPENLINEAGE_AIRFLOW_WHL=$(docker run --rm openlineage-airflow-base:latest sh -c "ls /whl/openlineage*.whl")
+OPENLINEAGE_AIRFLOW_WHL_ALL=$(docker run --rm openlineage-airflow-base:latest sh -c "ls /whl/*")
+
+# Add revision to requirements.txt
+cat > requirements.txt <<EOL
```
Comment (pawel-big-lebowski) [inline integration/airflow/scripts/run-dev-airflow.sh:47]: The less extra requirements to run tests, the better. 
Please consider, perhaps in an extra PR, removing those requirements and make use of a list available in `setup.py`.

**Review by pawel-big-lebowski — APPROVED** [review]
Added some comment for consideration.
:100: 👍

**Pull request merged**